server/controllers/project.controller/getProjectsForUser.js

```diff
@@ -17,16 +17,26 @@ function getProjectsForUserName(username) {
         reject(new UserNotFoundError());
         return;
       }
-
-      Project.find({ user: user._id })
+      const isUserAuthorized = authenicatedUser && username === authenicatedUser.username;
+      let query;
+      if (!isUserAuthorized) {
```

Comment:
I would remove this whole `if` to just make it a `const` since the whole point of the if is toggling one field to false, why not just make it true if the user is authorised and false if not which is basically what you're doing anyway:

```js
const query = {
      user: user._id,
      isPrivate: isUserAuthorized
};
```

Reply:
Exactly 😄

Reply:
I agree with you @Khensura21, that I don't think @jamesrweb's suggestion works for the behavior we want!

Reply:
But the condition is just saying if the user is authorised then the isPrivate will be false, otherwise it should be true, right? Seems strange to me to have a conditional to remove a key instead of flipping it’s value, or?

Reply:
This code is a MongoDB query, not setting or unsetting any values. Taking a close look at this again, I think there's actually a bug that's probably my fault from when we were working on this together. On line 25, I think we actually want to return all sketches where `isPrivate` is false or undefined, therefore (i think!!! definitely test this) lines 23-26 should be:
```js
        query = {
          user: user._id,
          $or: [
                { isPrivate: false },
                { isPrivate: { $exists: false } }
           ]
        };
```

Reply:
Fair enough, sounds fair with that being the case, good to know! 😁

Reply:
Sounds good. I will look into this and let you know!

Reply:
Sounds good. I will check this out and let you know!